test/services/nextPayment.service.test.js

```diff
@@ -0,0 +1,153 @@
+import mongoose from 'mongoose';
+import { expect, sinon } from '../config';
+import NextPayment from '../../server/models/nextPayment.model';
+import Offer from '../../server/models/offer.model';
+import Transaction from '../../server/models/transaction.model';
+import { generateNextPaymentDate } from '../../server/services/nextPayment.service';
+import OfferFactory from '../factories/offer.factory';
+import NextPaymentFactory from '../factories/nextPayment.factory';
+import TransactionFactory from '../factories/transaction.factory';
+
+describe('NextPayment Service', () => {
+  const offer = OfferFactory.build(
+    {
+      totalAmountPayable: 100000,
```

Comment:
Lets make this 10 million so that we don't easily exhaust the price with a single transaction
```suggestion
      totalAmountPayable: 10_000_000,
```